### PR TITLE
feat: add slug feature and anthropic enum attribute

### PR DESCRIPTION
**Pull request by jasonirvine76**
### Changes Made:

- Added slug Field:

Introduced a new slug attribute in the Bot model to ensure each bot has a unique, URL-friendly identifier.
Implemented find_bot_by_slug:

Developed a new method in the BotService class to retrieve bots using their slug. This method enhances the user experience by allowing easy access to specific bots without needing to use their ID.

- Updated Migration File:

Modified the database migration file to include the new slug column in the bots table, ensuring proper schema alignment with the updated model.

- Added Anthropic Enum

Introduced a new enum attribute for bot model

### Tests:

- Added unit tests to verify the functionality of the new slug field and the find_bot_by_slug method, ensuring that both create and retrieve operations work as expected.
- Adapt all tests for added slug field

### CODE REVIEW

File: migrations/20241010142723_add_slug_field_and_anthropic_enum.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+-- Add value to enum type: "model_engines"
+ALTER TYPE "public"."model_engines" ADD VALUE 'ANTHROPIC';
+-- Modify "bots" table
+ALTER TABLE "public"."bots" ADD COLUMN "slug" character varying(255) NULL;
```
Comment (Veivel) [inline migrations/20241010142723_add_slug_field_and_anthropic_enum.sql:4]: Does this migration file handle existing entries by allowing the column to be null, or is it genuinely optional?

**Comment by jasonirvine76** [comment]
> Sorry for the CI mishap. To break it down, the CI check for changed migration files did not handle the parsing of a JSON array string correctly. This caused `jq` to misinterpret the following string as a number instead of array `["migrations/20241010142723_add_slug_field_and_anthropic_enum.sql","migrations/atlas.sum"]`. Here is the reference for the issue [jqlang/jq#501](https://github.com/jqlang/jq/issues/501)
> 
> Can you merge the current change from main to this PR to accommodate the fix @jasonirvine76

Thanks for the explanation. I have merged the current changes to the branch.

**Comment by nayyara-airlangga** [comment]
Looks like the problem has been resolved 🚀

**Comment by TakeMeGH** [comment]
In this version, will the slug be filled in by the user and not auto-generated? If so, how do we ensure the slug is unique? Does the user need to provide a different slug if the slug already exists?

**Comment by jasonirvine76** [comment]
> In this version, will the slug be filled in by the user and not auto-generated? If so, how do we ensure the slug is unique? Does the user need to provide a different slug if the slug already exists?

For now, the slug must be filled in by the user. In the future, I'll make the slug auto-generated. Yes, the user needs to provide a different slug if the slug already exists in the database. We ensure that the slug is unique, so the user can't input the same slug.

File: migrations/20241010142723_add_slug_field_and_anthropic_enum.sql (hunk 1 above)
Comment (jasonirvine76) [reply migrations/20241010142723_add_slug_field_and_anthropic_enum.sql:4]: Yes, this migration handles existing entries by allowing the column to be NULL for now. Later, once we auto-generate the slug, I’ll change it to NOT NULL.

**Comment by Insta-x** [comment]
When the user input an existing slug, does the system respond with an error informing that the slug already existed?

**Comment by Insta-x** [comment]
> > In this version, will the slug be filled in by the user and not auto-generated? If so, how do we ensure the slug is unique? Does the user need to provide a different slug if the slug already exists?
> 
> For now, the slug must be filled in by the user. In the future, I'll make the slug auto-generated. Yes, the user needs to provide a different slug if the slug already exists in the database. We ensure that the slug is unique in DB, so the user can't input the same slug.

I just want to add a suggestion. Maybe keep the option for the user to make their own slug. This way, the user can customize the slug if they wanted. If they don't add their own slug, then use the autogeneration for the slug.

**Comment by jasonirvine76** [comment]
> When the user input an existing slug, does the system respond with an error informing that the slug already existed?

Yes, I’ll add that. I will create the check for duplicate slugs later.

**Pull request merged**